woocommerce/woocommerce-android
# Order List redesign - Step 1: Added TabLayout to the order list

Fixes #1399. This PR completes the first step of the Stats Improvements Project as defined in the master thread [here](https://github.com/woocommerce/woocommerce-android/issues/1398). 

### Changes
- Added a new class `OrderListView` that handles the order list related logic such as populating the `RecyclerView`, displaying empty view.
- Added `TabLayout` with `Processing` and `All Orders` tab and anchor the `TabLayout` to the top of the view when scrolled.
- Added logic to remember the last selected tab and default to that when order list screen is loaded.

### Notes
- The filter option will be removed in a separate PR when addressing #1400. 
- The search revamp will take place in a separate PR when addressing #1400.  
- Requires a design review from @kyleaparker before PR can be merged.
- **Note that this PR is to be merged to a feature branch.**

### Testing
- Run `OrderListItemTest`
- Test marking an order as complete from the `Processing` tab.
- Test changing an order status from the `All Orders` tab.
- Test pagination working as expected for both tabs.
- Search should work as it is currently.

### Screenshots
<img width="300" src="https://user-images.githubusercontent.com/22608780/64146200-a64c8080-ce39-11e9-9eb8-8a022d59ba1a.png"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/64146202-a8aeda80-ce39-11e9-85df-a92d41b84970.png">

#### Empty View
<img width="300" src="https://user-images.githubusercontent.com/22608780/64146226-c11ef500-ce39-11e9-9f9f-cf5c057b6070.png">

Update release notes:
